<?php

namespace Flikore\Validator\Exception;

/**
 * Generated by PHPUnit_SkeletonGenerator 1.2.0 on 2014-03-02 at 02:14:13.
 */
class ValidatorExceptionTest extends \PHPUnit_Framework_TestCase
{

    /**
     * @var ValidatorException
     */
    protected $object;

    /**
     * Sets up the fixture, for example, opens a network connection.
     * This method is called before a test is executed.
     */
    protected function setUp()
    {
        $this->object = new ValidatorException;
    }

    /**
     * Tears down the fixture, for example, closes a network connection.
     * This method is called after a test is executed.
     */
    protected function tearDown()
    {
        
    }

    public function testSetAndGetErrors()
    {
        $errs = array(
            'aa' => new ValidatorException('aa'),
            'bb' => new ValidatorException('bb'),
        );

        $this->object->setErrors($errs);

        $this->assertEquals($errs, $this->object->getErrors());
    }

    public function testSetAndGetError()
    {
        $errA = new ValidatorException('aa');
        $errB = new ValidatorException('bb');

        $this->object->setError($errA, 'aa');

        $this->assertEquals($errA, $this->object->getError('aa'));
        $this->assertNotEquals($errB, $this->object->getError('aa'));
    }

}
